pandas/core/frame.py

```diff
@@ -4049,33 +4049,96 @@ def _compare(a, b):
 
     def combine(self, other, func, fill_value=None, overwrite=True):
         """
-        Add two DataFrame objects and do not propagate NaN values, so if for a
-        (column, time) one frame is missing a value, it will default to the
-        other frame's value (which might be NaN as well)
+        Perform series-wise combine with `other` DataFrame using given `func`.
+
+        Combines `self` DataFrame with `other` DataFrame using `func`
+        to merge columns. The row and column indexes of the resulting
```

Comment:
merge is a very specific word and would rather not use it here. element-wise combine is pretty descriptive